### DISCUSSION TITLE
Version selector with index.md that uses a template

### DISCUSSION THREAD

**Discussion by sriumcp** [squidfunk/mkdocs-material#2531] [Q&A] [answered]
Thanks a lot for the super awesome Material for MkDocs. My question pertains to version selection, and in particular, this version selection example. 

![version selection example](https://squidfunk.github.io/mkdocs-material/assets/screenshots/versioning.png)

I forked the *squidfunk/mkdocs-material* repo, and added the version provider stanza in `mkdocs.yml`, and created `mike` versions. `mike` versioning works as expected since I am able to navigate between the different versions when  I `mike serve` the resultant site. However, the version selector does not show up when I `mike serve`.

Any tips on how I may get this to work? 

In particular, I notice that index.md has a template front matter configuration as follows. Is there a way I can get version selection to work even with this configuration?

```
---
template: overrides/home.html
---
```

[This example](https://squidfunk.github.io/mkdocs-material-example-versioning/latest/) on the other hand does work for me exactly as documented. I noticed that the `index.md` file does not have a template front-matter field here though.

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. Looping in @jimporter - AFAIK, `mike serve` serves from git, right? This would mean that you’d need to deploy your versions to be able to serve them. IIRC this is a design decision of `mike`.

Regarding the partials: as long as you don’t override and fiddle with the header partial in a custom template/override, the selector should show up.

**Reply by jimporter** [reply]
> `mike` versioning works as expected since I am able to navigate between the different versions when I `mike serve` the resultant site. However, the version selector does not show up when I `mike serve`.

I'm a bit confused; I assume the latter reference to `mike serve` is a typo. Did you mean `mkdocs serve` there?

As for your question @squidfunk, yes. `mike serve` serves pages from the latest revision of the `gh-pages` branch (or wherever your HTML docs are located), so they need to have been deployed first.

**Reply by sriumcp** [reply]
I did mean `mike serve`. The versioning / navigation using gh-pages branch worked for me.

Also, I am happy to confirm that version selector is also working for me now. It turns out that the issue did have to do with partials, which I was playing with. If I leave the header partial intact as  @squidfunk mentioned above, everything is working great as documented.
